### PR TITLE
Feature CMR-9539 - Update the coll-cache from standard redis to new Hash Map Redis 

### PR DESCRIPTION
**Pull request by jceaser**
Moving the code over to the new hash map redis cache and breaking up the data into two different hashes within that cache to limit the request data being transmitted over the "wire".

### CODE REVIEW

File: search-app/src/cmr/search/services/acls/collections_cache.clj [hunk 1]
```diff
@@ -88,6 +90,7 @@
                                          (for [{:keys [provider-id EntryTitle] :as coll}
                                                collections]
                                            [[provider-id EntryTitle] coll]))]
+    (debug "fetch-collections-map count" (count collections))
     ;; We could reduce the amount of memory here if needed by only fetching the collections that
     ;; have granules.
     {:by-concept-id by-concept-id
```
Comment (eereiter) [inline search-app/src/cmr/search/services/acls/collections_cache.clj:96]: Should we write a ticket to reduce the amount of memory by only fetching the collections that have granules?  Do we need the collections in this cache that don't have granules?

File: search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj [hunk 2]
```diff
@@ -91,7 +91,7 @@
         old-mem-cache (get-in context [:system :caches :collections-for-gran-acls])
         context (assoc-in context [:system :caches :collections-for-gran-acls] rcache)
         data (.get-value old-mem-cache :collections)]
-    (.set-value rcache :collections-for-gran-acls data)
+    (.set-values rcache :collections-for-gran-acls data)
```
Comment (eereiter) [inline search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj:94]: Is .set-values a short cut way of writing hash-cache/set-values ?

Comment (eereiter) [inline search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj:94]: Why did you leave old-mem-cache in the test?  Can you remove this and just set up the redis cache as the old-mem-cache (replace old-mem-cache)?  All you do with it is setup the memory cache and then take its contents and put it into the redis cache and then not use it anymore.

**Review by eereiter — APPROVED** [review]
It looks like the search app is loading in the cache instead of bootstrap.  I thought that at the beginning of the PI we were going to put the loading of the cache into bootstrap so that only 1 task loads the cache.  I saw this with a couple of other cache changes as well.  I think we should write a ticket to put the job scheduler of loading this cache into bootstrap.

File: search-app/src/cmr/search/services/acls/collections_cache.clj (hunk 1 above)
Comment (jceaser) [reply search-app/src/cmr/search/services/acls/collections_cache.clj:96]: that comment was already there, I did not change the logic of what was going to be cached

File: search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj (hunk 2 above)
Comment (jceaser) [reply search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj:94]: I was trying to keep the code as similar as possible. I can switch over to the other syntax.

File: search-app/src/cmr/search/services/acls/collections_cache.clj [hunk 3]
```diff
@@ -98,34 +101,40 @@
   to keep the cache fresh. This will throw an exception if there is a problem fetching collections. The
   caller is responsible for catching and logging the exception."
   [context]
-  (let [cache (cache/context->cache context cache-key)
+  (let [cache (hash-cache/context->cache context cache-key)
         collections-map (fetch-collections-map context)]
-    (cache/set-value cache cache-key (clj-times->time-strs collections-map))))
+    (doseq [[coll-key coll-value] collections-map]
+      (debug (format "collections-cache refresh-cache working on %s" coll-key))
```
[the remark this answers is not in the document]
Comment (jceaser) [reply search-app/src/cmr/search/services/acls/collections_cache.clj:107]: removing comment

File: search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj (hunk 2 above)
Comment (eereiter) [reply search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj:94]: you don't need to change over, I was just wondering.

Comment (jceaser) [reply search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj]: I don't think I even need this code, I removed it and tests pass locally, testing on bamboo now.

Comment (jceaser) [reply search-app/test/cmr/search/test/unit/services/acls/granule_acls.clj]: switched to the clojure syntax as suggested.

**Pull request merged**